### PR TITLE
Only allow a single instance of shuffleboard to launch

### PR DESCRIPTION
**Pull request by AustinShalit**

### CODE REVIEW

File: app/src/main/java/edu/wpi/first/shuffleboard/app/Shuffleboard.java [hunk 1]
```diff
@@ -40,12 +42,21 @@ public void init() throws Exception {
...
+      });
+
+      primaryStage.setScene(new Scene(mainPane));
+      primaryStage.setMinWidth(640);
+      primaryStage.setMinHeight(480);
+      primaryStage.setWidth(Screen.getPrimary().getVisualBounds().getWidth());
+      primaryStage.setHeight(Screen.getPrimary().getVisualBounds().getHeight());
+      primaryStage.show();
+    } catch (AlreadyLockedException e) {
+      JUnique.sendMessage("alreadyRunning", "true");
```
Comment (nightpool) [inline app/src/main/java/edu/wpi/first/shuffleboard/app/Shuffleboard.java]: sendMessage takes appId, which we're passing earlier as `getClass().getCanonicalName()`

**Comment by nightpool** [comment]
Why choose start over init? Shouldn't we abandon as early as possible if we know we don't need to load resources/layouts/etc?

File: app/src/main/java/edu/wpi/first/shuffleboard/app/Shuffleboard.java (hunk 1 above)
Comment (AustinShalit) [inline app/src/main/java/edu/wpi/first/shuffleboard/app/Shuffleboard.java]: Right

**Comment by AustinShalit** [comment]
>Why choose start over init? Shouldn't we abandon as early as possible if we know we don't need to load resources/layouts/etc?

I want the functionality of bringing the primary stage of the application that was already open to the front.  That object does not exist until start has been called.

**Comment by nightpool** [comment]
hmm. isn't that just

```java 
(message -> {
    if (mainPane != null) {
        mainPane.getStage().bringToFront()
    }
})
```

?

**Comment by AustinShalit** [comment]
Updated.

**Pull request merged**